### DISCUSSION TITLE
Truncating Select displayed string

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#2464] [Ideas / Feature Requests]
Consider I use Select with a list of options, some of which could be very long: `ui.select({1: 'short option', 2: 'a very very very very long option'})`.
Is it possible to use the full option label in the dropdown list, but once the option is selected, display a short version (like `short option` is unchanged, but `a very very very very long option` is displayed as `long option`)?

I found that `my_select.props(f'display-value="{my_value}"')` works, but only when new_value_mode is None.

It seems that https://quasar.dev/vue-components/select#the-display-value does what I need, but I can't wrap my head around how to use it from Python.

**Comment by me21** [comment]
Using that link and https://github.com/zauberzeug/nicegui/blob/main/examples/slots/main.py as a reference, I came up with a basic example for Python:

```py
surname_select.add_slot('selected-item', r'''
    <q-chip>
        Selected option: {{props.opt}}
    </q-chip>
''')
```

It seems I can access the "scope" from `selected-item` slot description as "props" here, including `props.opt` - the selected option. But can I add a Python UI control in that slot which can access that without writing inline Javascript code?

In addition, it doesn't work when I enable input by `new_value_mode='add'` when creating ui.select.

**Comment by me21** [comment]
Found I have to `.props(remove='hide-selected')` on the select control. Then it will display my `selected-item` slot. 

But a question about putting a Python NiceGUI control in that slot which could access `props` remains, although my immediate problem is solved.

**Reply by falkoschindler** [reply]
Hi @me21,
I'm glad to hear that you've found a solution.
Regarding slots with NiceGUI elements, I still don't know how this could be accomplished, see https://github.com/zauberzeug/nicegui/discussions/1138#discussioncomment-6383197.

**Reply by me21** [reply]
Some slots with NiceGUI elements do work, e.g. it's possible to put an icon in the slot. I think that works.

For more complex scenarios, I think we need a way to export Python function to call it from JavaScript side. It's like `run_javascript`, only vice versa. Then NiceGUI could probably build on this to provide a generic slot contents which are customised by calling Python.
